src/SocketContext.jsx

```diff
@@ -0,0 +1,76 @@
+import React, { createContext, useContext, useEffect, useState } from 'react';
+import io from 'socket.io-client';
+import { useAuth } from './AuthProvider';
+import NotificationModal from '../src/components/common/notificationModal';
+
+const SocketContext = createContext();
+
+export const SocketProvider = ({ children }) => {
+  const { userData } = useAuth();
+  const [socket, setSocket] = useState(null);
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [notificationMessage, setNotificationMessage] = useState('');
+
+  useEffect(() => {
+    if (!userData) {
```

Comment:
If the lack of communication to the user here is deliberate then that's fine. Since, we probably only want to send notifications when they are available. Otherwise we would need a more user friendly way of handling this.

Reply:
Sending notifications to users who are not online will be done through email. 
It will be a separate PR too.